symbolic_pymc/tensorflow/printing.py

```diff
@@ -81,11 +127,11 @@ def _(obj, printer):
     _tf_dprint(prefix, printer)
     if len(obj.op.inputs) > 0:
         with printer.indented("|  "):
-            if obj not in printer.printed_subgraphs:
+            if obj in printer.printed_subgraphs and printer.depth_index is None:
```

Comment:
Is this disabling the do-not-reprint-already-printed-subgraphs logic when limits are specified?  If so, that shouldn't happen.  

If the `printer.depth_index` condition is there because subgraphs that occur before the desired range are being added to `printer.printed_subgraphs` and, as a result, don't get printed in the desired depth range (i.e. when that range is at a greater depth), then that's a clear sign that this logic isn't correct.

It might make much more sense to put this do-not-reprint-already-printed-subgraphs logic in `TFlowPrinter.println` and modulate in accordance with `TFlowPrinter.depth_index` and `TFlowPrinter.depth_count` when subgraphs/terms are added to `printer.printed_subgraphs`&mdash;in the same way you could directly and concisely determine `TFlowPrinter.indentation` in `TFlowPrinter.indented`.

No matter what, this PR needs a test for depth limits + repeated graph elements.

Reply:
In the latest commit this no longer disables the do-not-reprint-already-printed-subgraphs logic when limits are specified.

Reply:
In the latest commit I added a new method that adds to `printed_subgraphs`. Adding do-not-reprint-already-printed-subgraphs  logic to `TFlowPrinter.println` runs into an issue since both `tensors` and `op` objects get passed.

Reply:
Added tests for both in the latest commit.